src/_P077_CSE7766.ino

```diff
@@ -84,11 +86,21 @@ boolean Plugin_077(uint8_t function, struct EventStruct *event, String& string)
       break;
     }
 
+    case PLUGIN_GET_DEVICEVALUECOUNT:
+    {
+      event->Par1 = P077_NR_OUTPUT_VALUES;
+      success     = true;
+      break;
+    }
+
     # if FEATURE_MQTT_DISCOVER
     case PLUGIN_GET_DISCOVERY_VTYPES:
     {
-      event->Par1 = static_cast<int>(Sensor_VType::SENSOR_TYPE_NONE); // Not yet supported
-      success     = true;
+      for (uint8_t i = 0; i < event->Par5; ++i) {
+        const uint8_t choice = PCONFIG(i + P077_QUERY1_CONFIG_POS);
```

Comment:
Maybe also add some check for i not getting out of bounds.
```c++
for (uint8_t i = 0; i < event->Par5 && (i < (8 - P077_QUERY1_CONFIG_POS)); ++i) 
```

Reply:
`Par5` is set using a call with function `PLUGIN_GET_DEVICEVALUECOUNT` (by calling function `getValueCountForTask()`), so should not get out of range.